pkg/sbom/cyclonedx/unmarshal.go

```diff
@@ -409,9 +410,31 @@ func toTrivyCdxComponent(component cdx.Component) ftypes.Component {
 }
 
 func getPackageName(typ string, component cdx.Component) string {
-	// Jar uses `Group` field for `GroupID`
-	if typ == packageurl.TypeMaven && component.Group != "" {
-		return fmt.Sprintf("%s:%s", component.Group, component.Name)
+	if typ == packageurl.TypeMaven {
+		return convertMavenPackage(component)
 	}
 	return component.Name
 }
+
+func convertMavenPackage(component cdx.Component) string {
+	// Check if both Group and Name are present
+	if component.Group != "" {
+		return fmt.Sprintf("%s:%s", component.Group, component.Name)
+	} else if component.PackageURL != "" {
+		// Split the package into its parts
+		parts := strings.Split(component.PackageURL, "/")
+
+		group := parts[1]
+
+		// Get FullName with Version
+		nameWithVersion := parts[len(parts)-1]
+
+		// Remove the Version from the package
+		nameWOVersion := strings.Split(nameWithVersion, "@")[0]
+
+		return fmt.Sprintf("%s:%s", group, nameWOVersion)
+		// In case we don't have Group or PURL
```

Comment:
We already parsed `purl` here - https://github.com/aquasecurity/trivy/blob/14d3f0242d3ab2a24e8982ab0f818524c263b1a1/pkg/sbom/cyclonedx/unmarshal.go#L347

We can use this.

Reply:
> We already parsed `purl` here -
> 
> https://github.com/aquasecurity/trivy/blob/14d3f0242d3ab2a24e8982ab0f818524c263b1a1/pkg/sbom/cyclonedx/unmarshal.go#L347
> 
> We can use this.

Great idea, thanks! Made some changes